nondex-maven-plugin/src/main/java/edu/illinois/nondex/plugin/CleanSurefireExecution.java

```diff
@@ -120,10 +124,26 @@ public void run() throws MojoExecutionException {
         }
     }
 
-    protected void setupArgline() {
-        Logger.getGlobal().log(Level.FINE, "Running clean surefire.");
+    protected void setupArgline(Xpp3Dom configNode) {
+        String argLineToSet =  this.configuration.toArgLine();
+        boolean added = false;
+        for (Xpp3Dom config : configNode.getChildren()) {
+            if ("argLine".equals(config.getName())) {
+                Logger.getGlobal().log(Level.INFO, "Adding argLine to existing one");
+                String current = config.getValue();
+
+                config.setValue(argLineToSet + " " + current);
+                added = true;
+                break;
+            }
+        }
+        if (!added) {
+            Logger.getGlobal().log(Level.INFO, "Adding argline to newly created one");
+            configNode.addChild(this.makeNode("argLine", argLineToSet));
+        }
+
         this.mavenProject.getProperties().setProperty("argLine",
-                this.originalArgLine + " " + this.configuration.toArgLine());
+                this.originalArgLine + " " + argLineToSet);
```

Comment:
Comments needed here: how are `this.originalArgLine` and `current` different? Why the need to append here again?